### PR TITLE
Adopt new MPSNNOptimizer API for updating conv and batch-norm weights

### PR DESCRIPTION
**Pull request by nickjong**
Deeper fix for #742: adopt the newer convenience API for updating our data sources

### CODE REVIEW

File: src/unity/toolkits/tcmps/mps_weight.mm [hunk 1]
```diff
@@ -9,7 +9,116 @@
...
+
+@property (nonatomic, readonly) MPSVector *weightMomentumVector;
+@property (nonatomic, readonly) MPSVector *weightVelocityVector;
+@property (nonatomic, readonly) MPSVector *biasMomentumVector;
+@property (nonatomic, readonly) MPSVector *biasVelocityVector;
+
+@end
+
+// Class extension for TCMPSBatchNormData internal implementation
+@interface TCMPSBatchNormData ()
```
Comment (gustavla) [inline src/unity/toolkits/tcmps/mps_weight.mm]: Should we perhaps call this `TCMPSBatchNormWeights`? Not sure why we have convolution *weights* but batch norm *data*. I suppose the BN gammas and betas are what we normally consider to be weights, but the moving mean/var don't typically go by this name. However, at the end of the day, during inference, they are weights in the network, so the distinction is too subtle to warrant the -Data name.

Comment (nickjong) [reply src/unity/toolkits/tcmps/mps_weight.mm]: Good call! I went ahead and renamed to `TCMPSBatchNormWeights`
